frontend/src/Sidebar.tsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import './Sidebar.css';
+import { LoginManager } from "./LoginManager";
+
+export interface PageComponents {
+    components: PageComponent[]
+    loginManager: LoginManager
+}
+
+export interface PageComponent {
+    id: number
+    title: string
+    page: JSX.Element
+    icon: JSX.Element
+    selected?: boolean
+}
+
+/** Menu component which holds the different pages and makes them accesible. */
+export class Sidebar extends React.Component<PageComponents, PageComponents>{
+    constructor(props: PageComponents, state: PageComponents) {
+        super(props);
+        this.state = props;
+    }
+
+    /** Set this page active. Remove the old page and select the new one to be rendered. */
+    private setActivePage(pageId: number) {
+        var newComponents = [];
+        for (let c = 0; c < this.state.components.length; c++) {
```

Comment:
Could be simplified using `for..in` loop:
```suggestion
        for (let c in this.state.components) {
```

Reply:
Then I get a linter error on the next line complayining that c would be of the type string

Reply:
For some reason, the index is a string instead of a number. But if you use `for..of` instead, you'll get the actual value instead of the index.